webui/react/src/components/CheckpointModal.tsx

```diff
@@ -0,0 +1,118 @@
+import { Modal } from 'antd';
+import React, { useCallback, useMemo } from 'react';
+
+import Badge, { BadgeType } from 'components/Badge';
+import { CheckpointDetail, CheckpointStorageType, ExperimentConfig, RunState } from 'types';
+import { formatDatetime } from 'utils/date';
+import { humanReadableBytes, humanReadableFloat } from 'utils/string';
+
+import css from './CheckpointModal.module.scss';
+import Link from './Link';
+
+interface Props {
+  checkpoint: CheckpointDetail;
+  config: ExperimentConfig;
+  onHide?: () => void;
+  show?: boolean;
+}
+
+const getStorageLocation = (config: ExperimentConfig, checkpoint: CheckpointDetail): string => {
+  const hostPath = config.checkpointStorage?.hostPath;
+  const storagePath = config.checkpointStorage?.storagePath;
+  let location = '';
+  switch (config.checkpointStorage?.type) {
+    case CheckpointStorageType.AWS:
+      location = `s3://${config.checkpointStorage.bucket || ''}`;
+      break;
+    case CheckpointStorageType.GCS:
+      location = `gs://${config.checkpointStorage.bucket || ''}`;
+      break;
+    case CheckpointStorageType.SharedFS:
+      if (hostPath && storagePath) {
+        location = storagePath.startsWith('/') ?
+          `file://${storagePath}` : `file://${hostPath}/${storagePath}`;
+      } else if (hostPath) {
+        location = `file://${hostPath}`;
+      }
+      break;
+  }
+  return `${location}/${checkpoint.id}`;
+};
+
+const renderRow = (label: string, content: React.ReactNode): React.ReactNode => {
+  return (
+    <div className={css.row} key={label}>
+      <div className={css.label}>{label}</div>
+      <div className={css.content}>{content}</div>
+    </div>
+  );
+};
+
+const renderResource = (resource: string, size: string): React.ReactNode => {
+  return (
+    <div className={css.resource} key={resource}>
+      <div className={css.resourceName}>{resource}</div>
+      <div className={css.resourceSpacer} />
+      <div className={css.resourceSize}>{size}</div>
+    </div>
+  );
+};
+
+const CheckpointModal: React.FC<Props> = ({ config, checkpoint, onHide, show }: Props) => {
+  const state = checkpoint.state as unknown as RunState;
+
+  const handleHide = useCallback(() => {
+    if (onHide) onHide();
+  }, [ onHide ]);
+
+  const totalSize = useMemo(() => {
+    const total = Object.values(checkpoint.resources).reduce((acc, size) => acc + size, 0);
+    return humanReadableBytes(total);
+  }, [ checkpoint.resources ]);
+
+  const resources = useMemo(() => {
```

Comment:
would be cool if we could use something like https://ourcodeworld.com/articles/read/1068/how-to-render-properly-a-directory-structure-treeview-in-prism-js or parts of https://github.com/mperitz/react-filetree-electron for the resource view but that's a bigger task

Reply:
I'll create a ticket for this so we can keep track of it for the future.